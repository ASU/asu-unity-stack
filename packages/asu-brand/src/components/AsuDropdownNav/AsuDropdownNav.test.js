import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ASUDropdownNav from "./AsuDropdownNav";

const titems = [
  {
    text: "Item 0",
    href: "https://asu.edu",
  },
  {
    text: "Item 1",
    href: "https://asu.edu",
  },
  {
    text: "Item 2",
    href: "https://asu.edu",
  },
  {
    text: "Item 3",
    href: "https://asu.edu",
  },
  {
    text: "Item 4",
    href: "https://asu.edu",
  },
  {
    text: "Item 5",
    href: "https://asu.edu",
  },
  {
    text: "Item 6",
    href: "https://asu.edu",
  },
  {
    text: "Item 7",
    href: "https://asu.edu",
  },
  {
    text: "Item 8",
    href: "https://asu.edu",
  },
  {
    text: "Item 9",
    href: "https://asu.edu",
  },
];

describe("The AsuDropdownMenu", () => {
  it("renders the component", () => {
    const { container } = render(
      <ASUDropdownNav {...{ text: "ASU Home", items: titems }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("drops down when button clicked", async () => {
    const { getByText, findAllByText } = render(
      <ASUDropdownNav {...{ text: "ASU Home", items: titems }} />
    );
    // Click button
    fireEvent.click(getByText("ASU Home"));

    // Wait for page to update with query text
    const items = await findAllByText(/Item [0-9]/);
    expect(items).toHaveLength(10);
  });
});
