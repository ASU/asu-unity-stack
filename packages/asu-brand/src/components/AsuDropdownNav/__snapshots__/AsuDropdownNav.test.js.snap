// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AsuDropdownMenu renders the component 1`] = `
<div
  class="asuNavItemTop"
  role="navigation"
  title="ASU Home"
>
  <span>
    ASU Home
  </span>
  <span
    class="icoSortDown"
  />
  <ul
    class="subMenu subSlideOut"
  >
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 0
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 1
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 2
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 3
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 4
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 5
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 6
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 7
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 8
      </a>
    </li>
    <li
      class="asuNavItem"
    >
      <a
        href="https://asu.edu"
        tabindex="0"
      >
        Item 9
      </a>
    </li>
  </ul>
</div>
`;
